### ISSUE TITLE
Bug: single run mode ignores the debugLevel flag

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#5887]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

In normal type-aware mode we pass a special set of compiler options to our program which allows us to turn on TS debug logging.

Call:
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-estree/src/create-program/createWatchProgram.ts#L265

Utility:
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-estree/src/create-program/shared.ts#L36-L47

But in single run mode we don't use our util, so we don't turn on debug logging when we should:
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-estree/src/create-program/useProvidedPrograms.ts#L66

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  parserOptions: {
    debugLevel: ['typescript'],
  },
};
```

**Comment by bradzacher** [comment]
turns out this doesn't matter - TS doesn't actually log anything unless you're using the watch host.
